### DISCUSSION TITLE
macOS: Help with your vote for Firefox AutoFill

### DISCUSSION THREAD

**Discussion by latvia234** [keepassium/KeePassium#385] [General]
Right now there is no native support for AutoFill inside Firefox.

If you see this message, could you please give your vote on the following page? https://connect.mozilla.org/t5/ideas/add-support-for-apple-s-autofill-system-on-mac/idi-p/5345

It really helps.

@keepassium could you also add [this link](https://connect.mozilla.org/t5/ideas/add-support-for-apple-s-autofill-system-on-mac/idi-p/5345) to [this article](https://keepassium.com/articles/how-to-setup-autofill/)? Because people who read that article could be searching how to enable AutoFill for Firefox. You could ask them to vote for native AutoFill support.

There's also another place where you can vote: https://bugzilla.mozilla.org/show_bug.cgi?id=1650212 (After login, go to the very top, expand "Details", there is a "vote"-Button.)

**Comment by keepassium** [comment]
I have added browser-feedback links to https://support.keepassium.com/kb/autofill-quick/#browsers

But I think this is a lost cause. 

1. Both Firefox and Chrome have their own in-built password managers. They are more interested in coercing users to their solutions, rather than opening door for Apple Keychain (mainly) and other solutions.
2. iOS 17 / macOS 14 fixed the problem partially, by including "AutoFill" option to context menu of text inputs. So now AutoFill works for any text field in any app, unless they specifically opt-out. The problem is, that AutoFill menu is hardwired to Apple Keychain.
3. Reportedly, iOS 18 / macOS 15 will open that context menu for third-party password managers, too. So browser support would become more of a "nice to have", further reducing the pressure on browser makers.

**Reply by latvia234** [reply]
> Reportedly, iOS 18 / macOS 15 will open that context menu for third-party password managers, too. So browser support would become more of a "nice to have", further reducing the pressure on browser makers.

Did not know about this. Thanks for your reply! Let's hope that they'll add support in iOS 18 / macOS 15.
